Source/WebCore/css/CSSCounterStyleDescriptors.cpp

```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
     return primitiveValue.stringValue();
 }
 
-static CSSCounterStyleDescriptors::Pad translatePadFromStyleProperties(const StyleProperties& properties)
+static CSSCounterStyleDescriptors::AdditiveSymbols translateAdditiveSymbolsFromStyleProperties(const StyleProperties& properties)
 {
-    auto pad = properties.getPropertyCSSValue(CSSPropertyPad).get();
-    if (!pad || !is<CSSValueList>(pad))
-        return CSSCounterStyleDescriptors::Pad();
+    auto additiveSymbols = properties.getPropertyCSSValue(CSSPropertyAdditiveSymbols);
+    if (!additiveSymbols)
+        return { };
+    auto* list = downcast<CSSValueList>(additiveSymbols.get());
+    if (!list)
+        return { };
+
+    CSSCounterStyleDescriptors::AdditiveSymbols result;
+    for (auto& additiveValue : *list) {
+        if (auto* additiveTuple = dynamicDowncast<CSSValuePair>(&additiveValue)) {
+            auto& weightValue = additiveTuple->first();
+            auto& symbolValue = additiveTuple->second();
+            auto& weight = downcast<CSSPrimitiveValue>(weightValue);
+            result.constructAndAppend(symbolToString(&symbolValue), weight.value<unsigned>());
```

Comment:
```suggestion
        if (auto* pair = dynamicDowncast<CSSValuePair>(&value)) {
            auto weight = downcast<CSSPrimitiveValue>(pair->first());
            auto symbol = symbolToString(&pair->second());
            result.constructAndAppend(symbol, weight.value<unsigned>());
```

Reply:
Should be `auto& weight`.

Reply:
I would just write `auto weight = downcast<CSSPrimitiveValue>(pair->first()).value<unsigned>();`

Reply:
yes, I've noticed the copy constructor is deleted. I'll follow the suggestion, thanks.